tests/datavolume_test.go

```diff
@@ -0,0 +1,63 @@
+package tests_test
+
+import (
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+
+	"kubevirt.io/kubevirt-ansible/tests"
+	kubev1 "kubevirt.io/kubevirt/pkg/api/v1"
+	ktests "kubevirt.io/kubevirt/tests"
+)
+
+// template parameters
+const rawDataVolumeVMFilePath = "tests/manifests/template/datavolume-vm.yml"
+const rawDataVolumeVMIFilePath = "tests/manifests/template/datavolume-vmi.yml"
+const rawDataVolumeFilePath = "tests/manifests/template/datavolume.yml"
+
+var _ = Describe("DataVolume Integration Test", func() {
+	var dataVolumeName, vmName, dstDataVolumeFilePath, url, dstVMIFilePath string
+
+	Context("Datavolume with VM", func() {
+		BeforeEach(func() {
+			dataVolumeName = "datavolume1"
+			vmName = "test-vm-i"
+			dstDataVolumeFilePath = "/tmp/test-datavolume-vm.json"
+			url = "https://download.cirros-cloud.net/0.4.0/cirros-0.4.0-x86_64-disk.img"
```

Comment:
We don't want to depend on or use a external url for image download. This impacts the reliability of the test.  Their server has to be available, and the url has to remain valid for our test to pass.

Instead, we should be deploying an http server of our own in the test environment that can serve up static images like this.  This is what both the CDI and KubeVirt functional tests do. You can find examples there.

Reply:
We are trying to avoid this because:
1. some of our testing images might really huge, so adding large static image into repo like [this](https://github.com/kubevirt/containerized-data-importer/pull/315) is not that ideal
2. we trying to stimulate real user behavior, so I believe most of user might fetching url externally even though it might has some network issuses.

adding @nellyc  here to discuss

Reply:
> some of our testing images might really huge, so adding large static image into repo like this is not that ideal

In KubeVirt CI we don't actually commit the disk image to the repo. We pull in the image once at built time (from the external url), put it in a container image with something like nginx to serve the file, and deploy it within the cluster as test infrastructure.

If we don't do it this way, we'll likely have timing issues with our functional tests where sometimes the import takes longer than expected causing transient failures.

Reply:
@shiywang  Here's the KubeVirt webserver's Dockerfile. https://github.com/kubevirt/kubevirt/blob/master/images/cdi-http-import-server/Dockerfile

We do it slightly different that CDI. CDI chose to include that tiny.iso file in the repo, but like your use case, our images are too large to put in the repo itself.

Reply:
it currently being blocked, I will resolve it in another pr